pkg/controller/core/localqueue_controller.go

```diff
@@ -105,34 +145,54 @@ func (r *LocalQueueReconciler) NotifyWorkloadUpdate(oldWl, newWl *kueue.Workload
 // +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=localqueues/finalizers,verbs=update
 
 func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	var queueObj kueue.LocalQueue
-	if err := r.client.Get(ctx, req.NamespacedName, &queueObj); err != nil {
+	var lq kueue.LocalQueue
+	if err := r.client.Get(ctx, req.NamespacedName, &lq); err != nil {
 		// we'll ignore not-found errors, since there is nothing to do.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	log := ctrl.LoggerFrom(ctx)
 	log.V(2).Info("Reconcile LocalQueue")
 
-	if ptr.Deref(queueObj.Spec.StopPolicy, kueue.None) != kueue.None {
-		err := r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, StoppedReason, localQueueIsInactiveMsg)
+	if ptr.Deref(lq.Spec.StopPolicy, kueue.None) != kueue.None {
+		err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, StoppedReason, localQueueIsInactiveMsg)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	var cq kueue.ClusterQueue
-	err := r.client.Get(ctx, client.ObjectKey{Name: string(queueObj.Spec.ClusterQueue)}, &cq)
-	if err != nil {
+	if err := r.client.Get(ctx, client.ObjectKey{Name: string(lq.Spec.ClusterQueue)}, &cq); err != nil {
 		if apierrors.IsNotFound(err) {
-			err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, "ClusterQueueDoesNotExist", clusterQueueIsInactiveMsg)
+			err = r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, "ClusterQueueDoesNotExist", clusterQueueIsInactiveMsg)
 		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	if meta.IsStatusConditionTrue(cq.Status.Conditions, kueue.ClusterQueueActive) {
-		err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionTrue, "Ready", "Can submit new workloads to localQueue")
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		if err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionTrue, "Ready", "Can submit new workloads to localQueue"); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+	} else {
+		if err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, clusterQueueIsInactiveReason, clusterQueueIsInactiveMsg); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
 	}
-	err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, clusterQueueIsInactiveReason, clusterQueueIsInactiveMsg)
-	return ctrl.Result{}, client.IgnoreNotFound(err)
+
+	if r.admissionFSConfig != nil {
```

Comment:
IIUC both `initializeAdmissionFsStatus` and `reconcileConsumedUsage` update the LQ status. Can we combine them? If this is non trivial, I think a follow up issue is ok.

Reply:
I'd rather do it in a follow-up. We need to have it initialized before we compare it with the interval

Reply:
sgtm, it is just a performance improvement. Though it is typically a good practice / idiomatic to collect all updates to a resource, and send the patch at the end of reconcile

Reply:
Will do in a follow-up then